tests/integration/markers.py

```diff
@@ -0,0 +1,8 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import pytest
+
+from . import juju_
+
+skip_juju_2_9 = pytest.mark.skipif(juju_.is_2_9, reason="Skips juju v2.9.x")
```

Comment:
Not exactly, the newer `self-signed-certificates` charm only accepts juju 3.1+. Therefore, asked Mykola here: https://chat.canonical.com/canonical/pl/eq13bogi438qppjn816x4yngac

To skip these tests. So, I am skiping in the end because they are juju 2.9, not that they have or not secrets.

In either case, I am using the same structure as mysql's: there is a `juju_.py` file that checks the version and a `markers.py` file that marks the `skip_2_9`.